src/ids.rs

```diff
@@ -0,0 +1,109 @@
+/// Maps u64 integers to a consecutive range of ids
+#[derive(Debug)]
+pub struct IdTable {
+    // map u64 id x to u32 by storing a sorted mapping table for each value of x / 2^32
+    data: Vec<Vec<(u32, u32)>>,
```

Comment:
Interesting approach. I would have thought that this is actually bigger than a hashmap by just looking at the u32 space.

Reply:
Ok, since you taking the higher bits as set address, and ids are usually contiguous, there will be much less sets than u32-space.